### PR TITLE
🎉 Stripe Source: Fix subscriptions stream to return all kinds of subscriptions (including expired and canceled)

### LINKED ISSUE #4374: Stripe Source - Expired/Cancelled Subscriptions Not Being Retrieved
## Expected Behavior
Stripe source should retrieve all subscription statuses for the Stripe subscription source. Capturing expiration/cancel/etc is essential to Stripe data integration

## Current Behavior
Appears that the connector is currently pulling only active subscriptions, which is the default request.
https://stripe.com/docs/api/subscriptions

## Logs
N/A

## Steps to Reproduce
1. Execute Stripe > Target Connection
2. Review Subscription ABC123 data, active
3. Customer cancels subscription ABC123
4. No subscription record is returned on the next execution for subscription ABC123

## Severity of the bug for you
High

## Airbyte Version
0.26.2-alpha

## Connector Version (if applicable)
0.1.12

### PR DESCRIPTION
**Pull request by annalvova05**
## What
Stripe source should retrieve all subscription statuses for the Stripe subscription source.

## How
- add "status" param in API request
- edited the README.md, updated invalid statements for tests
- fixed the stream schemas: `balance-transactions, bank_accounts, charges, customer_balance_transactions, customers, disputes, events, invoice_items, invoices, payouts, products, transfers`
- removed the `$ref` dependencies from schemas as they referred to `shared/_schema_.json`, which did not exist in the git repository, so the tests were failed as well
- added test for `Subscription` stream using the subset records of subscriptions stream to `acceptance-test-config.yaml`
- added "expected_records.json" example to `/docs/contributing-to-airbyte/building-new-connector/source-acceptance-tests.md`
 

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [x] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

**Comment by midavadim** [comment]
Change PR description and docs description to :
Stripe Source: retrieve all kinds of Subscriptions (including expired and canceled)

File: docs/integrations/sources/stripe.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ If you would like to test Airbyte using test data on Stripe, `sk_test_` and `rk_
 
 | Version | Date       | Pull Request | Subject |
 | :------ | :--------  | :-----       | :------ |
+| 0.1.14   | 2021-07-09 | [4669](https://github.com/airbytehq/airbyte/pull/4669) | Add status param for Subscriptions Stream |
```
Comment (midavadim) [inline docs/integrations/sources/stripe.md:67]: Change PR description and docs description to :
Stripe Source: retrieve all kinds of Subscriptions (including expired and canceled)

**Comment by midavadim** [comment]
Anna, let's try to add tests which will test output as well checking .

You need to 
1. create 'integration_tests/expected_records.txt" which contains output of read method  (but maybe  it should have only RECORD messages)

2. add additional basic_read test in acceptance-test-config.yml  file, like this:
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      validate_output_from_all_streams: yes
      expect_records:
        path: "integration_tests/expected_records.txt"
        extra_fields: no
        exact_order: no
        extra_records: yes

**Review by midavadim — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-stripe/CHANGELOG.md [hunk 2]
```diff
@@ -2,3 +2,5 @@
 
 ## 0.1.8
 Start using the new Source Acceptance tests
+## 0.1.14
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/CHANGELOG.md:5]: this file is not needed -- let's remove it and point to the docs in `docs/...`

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py [hunk 3]
```diff
@@ -211,10 +211,18 @@ def path(self, **kwargs):
...
     cursor_field = "created"
+    status = "all"
 
     def path(self, **kwargs):
         return "subscriptions"
 
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["created[gte]"] = stream_state.get(self.cursor_field)
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/source_stripe/source.py:222]: Done.

File: docs/integrations/sources/stripe.md (hunk 1 above)
Comment (bazarnov) [reply docs/integrations/sources/stripe.md]: Done.

File: airbyte-integrations/connectors/source-stripe/CHANGELOG.md (hunk 2 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/CHANGELOG.md]: Done.

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_subscriptions_statuses_records.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
...bscription=sub_HeFKfIvdgFh3qw"}, "latest_invoice": "in_1J8OZSIEn5WyEQxnqy1SOqzg", "livemode": false, "metadata": {"eligibleForTrial": "false"}, "next_pending_invoice_item_invoice": null, "pause_collection": null, "pending_invoice_item_interval": null, "pending_setup_intent": null, "pending_update": null, "plan": {"id": "plan_HDfijky2JcM0pm", "object": "plan", "active": true, "aggregate_usage": "sum", "amount": null, "amount_decimal": null, "billing_scheme": "tiered", "created": 1588637439, "currency": "usd", "interval": "month", "interval_count": 1, "livemode": false, "metadata": {}, "nickname": "free-overage", "product": "prod_H07MQuAFARZ6K6", "tiers": [{"flat_amount": null, "flat_amount_decimal": null, "unit_amount": 0, "unit_amount_decimal": "0", "up_to": 5000}, {"flat_amount": null, "flat_amount_decimal": null, "unit_amount": 0, "unit_amount_decimal": "0", "up_to": null}], "tiers_mode": "graduated", "transform_usage": null, "trial_period_days": null, "usage_type": "metered"}, "quantity": 1, "schedule": null, "start_date": 1594766545, "status": "active", "tax_percent": null, "transfer_data": null, "trial_end": 1596283200, "trial_start": 1595483518}, "emitted_at": 1626172757000}}
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/integration_tests/expected_subscriptions_statuses_records.json:2]: where is this used?

**Review by midavadim — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_subscriptions_statuses_records.json (hunk 4 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_subscriptions_statuses_records.json]: This subset of records is used in `source-acceptance-tests` for checking for expected_records in basic_read for subscription stream.

File: .github/workflows/test-command.yml [hunk 5]
```diff
@@ -124,8 +124,8 @@ jobs:
           SOURCE_SQUARE_CREDS: ${{ secrets.SOURCE_SQUARE_CREDS }}
           SOURCE_MARKETO_SINGER_INTEGRATION_TEST_CONFIG: ${{ secrets.SOURCE_MARKETO_SINGER_INTEGRATION_TEST_CONFIG }}
           SOURCE_RECURLY_INTEGRATION_TEST_CREDS: ${{ secrets.SOURCE_RECURLY_INTEGRATION_TEST_CREDS }}
-          STRIPE_INTEGRATION_TEST_CREDS: ${{ secrets.STRIPE_INTEGRATION_TEST_CREDS }}
-          STRIPE_INTEGRATION_CONNECTED_ACCOUNT_TEST_CREDS: ${{ secrets.STRIPE_INTEGRATION_CONNECTED_ACCOUNT_TEST_CREDS }}
+          SOURCE_STRIPE_CREDS: ${{ secrets.SOURCE_STRIPE_CREDS }}
+          # STRIPE_INTEGRATION_CONNECTED_ACCOUNT_TEST_CREDS: ${{ secrets.STRIPE_INTEGRATION_CONNECTED_ACCOUNT_TEST_CREDS }}
```
Comment (sherifnada) [inline .github/workflows/test-command.yml:128]: this should not be commented out because some tests should depend on it

File: airbyte-integrations/connectors/source-stripe/README.md [hunk 6]
```diff
@@ -49,10 +49,10 @@ and place them into `secrets/config.json`.
 
 ### Locally running the connector
 ```
-python main_dev.py spec
-python main_dev.py check --config secrets/config.json
-python main_dev.py discover --config secrets/config.json
-python main_dev.py read --config secrets/config.json --catalog sample_files/configured_catalog.json
+python main.py spec
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/README.md:52]: thanks for updating docs 👍🏼

File: airbyte-integrations/connectors/source-stripe/README.md [hunk 7]
```diff
@@ -61,17 +61,26 @@ To run unit tests locally, from the connector directory run:
...
+```
+To run your integration tests with docker
+
 ### Locally running the connector docker image
 
 #### Build
 First, make sure you build the latest Docker image:
 ```
-docker build . -t airbyte/source-stripe:dev
+docker build --no-cache . -t airbyte/source-stripe:dev
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/README.md:78]: why `--no-cache`?

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml [hunk 8]
```diff
@@ -5,34 +5,24 @@ tests:
   connection:
     - config_path: "secrets/config.json"
       status: "succeed"
-    - config_path: "secrets/connected_account_config.json"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:8]: this should not be removed as it's testing an important feature in this connector

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscriptions.json [hunk 9]
```diff
@@ -17,24 +16,36 @@
       "format": "date-time"
     },
     "items": {
-      "items": {
-        "type": ["null", "string"]
-      },
-      "type": ["null", "array"]
+      "type": ["null", "object"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscriptions.json:19]: so this schema was just incorrect?

File: docs/contributing-to-airbyte/building-new-connector/source-acceptance-tests.md [hunk 10]
```diff
@@ -67,6 +67,18 @@ Configuring all streams in the input catalog to full refresh mode verifies that
 |||x|
 ||||
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/source-acceptance-tests.md:69]: beautiful 👍🏼

File: tools/bin/ci_credentials.sh [hunk 11]
```diff
@@ -88,8 +88,7 @@ write_standard_creds source-slack-singer "$SLACK_TEST_CREDS"
 write_standard_creds source-smartsheets "$SMARTSHEETS_TEST_CREDS"
 write_standard_creds source-snowflake "$SNOWFLAKE_INTEGRATION_TEST_CREDS" "config.json"
 write_standard_creds source-square "$SOURCE_SQUARE_CREDS"
-write_standard_creds source-stripe "$STRIPE_INTEGRATION_TEST_CREDS"
-write_standard_creds source-stripe "$STRIPE_INTEGRATION_CONNECTED_ACCOUNT_TEST_CREDS" "connected_account_config.json"
```
Comment (sherifnada) [inline tools/bin/ci_credentials.sh:92]: connected account creds should remain

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml [hunk 12]
```diff
@@ -5,34 +5,24 @@ tests:
...
     - config_path: "integration_tests/invalid_config.json"
       status: "failed"
   discovery:
     - config_path: "secrets/config.json"
-    - config_path: "secrets/connected_account_config.json"
   basic_read:
+    # Test for subscriptions only with respect to subscription status in ["active","canceled"]
+    # If this test fails for some reason, please check the expected_subscriptions_statuses_records.json for valid subset of data.
     - config_path: "secrets/config.json"
-#     Reading a invoice_line_items stream takes too long and falls on timeout.
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:17]: why did you remove these test cases?

**Review by sherifnada — CHANGES_REQUESTED** [review]
Generally LGTM, thanks for updating with the new CDK version and removing old tests. But requesting changes because there are removed test cases which should still be there

File: airbyte-integrations/connectors/source-stripe/README.md (hunk 7 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/README.md:78]: locally, if you rebuild the docker image multiple times in a row it saves the cache and doesn't accept the minor changes, let's say changes in the schemas. So put this as it is, and it solved the issue with the docker image.

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscriptions.json (hunk 9 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/source_stripe/schemas/subscriptions.json:19]: yes

File: docs/contributing-to-airbyte/building-new-connector/source-acceptance-tests.md (hunk 10 above)
Comment (bazarnov) [reply docs/contributing-to-airbyte/building-new-connector/source-acceptance-tests.md:69]: I thought it should be there as an example, so thanks.

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml (hunk 12 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:17]: on the stage of acceptance tests they are failing, did you know that those tests are never get triggered, because previously this connector was on the `standard-source-test` not on `source-acceptence-test`, and as far as I checked the `acceptence-test-config.yaml` is only triggered with `source-acceptence-tests`.

I'll try to bring them back, and test again.

File: tools/bin/ci_credentials.sh (hunk 11 above)
Comment (bazarnov) [reply tools/bin/ci_credentials.sh:92]: ok

File: .github/workflows/test-command.yml (hunk 5 above)
Comment (bazarnov) [reply .github/workflows/test-command.yml:128]: ok

File: airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml (hunk 12 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:17]: The next schemas are not valid at this point and these should be edited corresponding to the real data types in the output:
`balance-transactions, bank_accounts, charges, customer_balance_transactions, customers, disputes, events, invoice_items, invoices, payouts, products, transfers`

Comment (sherifnada) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:17]: thanks!

Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/acceptance-test-config.yml:17]: Done. Tests are fixed, the timeout value of 3600 is set, because from time to time tests are passing too long.

**Review by sherifnada — APPROVED** [review]
